### PR TITLE
feat(api): add user profile data endpoints

### LINKED ISSUE #98: User profile get/update endpoints
Create endpoints for `UserProfileService`:

- [ ] GET `/api/v1/user/profile` gets `user` data (except salt & hashedPassword) plus interests list.
- [ ] PATCH `/api/v1/user/profile` to update basic info

### PR DESCRIPTION
**Pull request by tgrigoruk**
### Summary
Create API endpoints for getting and setting profile info for a logged in user.

Added endpoints to `UserService`:

- GET `/api/v1/user/:id?/profile` responds with the specified user's profile data (with some fields omitted), or the logged in user's data if id unused 

- PATCH `/api/v1/user/profile` updates specified fields, may return `{errors: [parsing errors...]}` on malformed data.

### Retrospective

Zod is pretty cool for parsing, I'm sure there's a lot more we can do with it. I'm using it as simply as necessary for now.

I'm not too happy with how I'm using response codes and error messages - we should determine some standards for these.

### Reviewer's Notes

Getting another user's profile may depend on friend status (ie not permissible if BLOCKED), which is not yet implemented.

### Testing

Hit up those endpoints with various form data and see what you get back. Dates can be funny. Note which fields can/cannot be edited - we may reconsider some of them.

### CODE REVIEW

File: apps/api/src/server/services/UserService.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  public async updateProfile(id: number, updateData: ProfileData) {
+    const results = await this.resources.db
+      .update(user)
+      .set(updateData)
+      .where(eq(user.id, id));
+
+    // affectedRows will be 1 even if no data was actually changed;
+    // changedRows would show actual db changes (but is deprecated)
+    return { updated: results[0].affectedRows === 1 };
```
Comment (tgrigoruk) [inline apps/api/src/server/services/UserService.ts:56]: I don't think the requester needs to know whether change was actually made...

File: apps/api/src/server/services/UserService.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    // TODO: use middleware to check authentication
+
+    this.router.get('/profile', async (req, res) => {
+      // TODO: use query param to get profile of specified user (if friend)
+      const id  = req.session.userId;
+      if (!id) {
+        res.status(401).json({ error: 'Not authenticated' });
+        return;
+      }
```
Comment (tgrigoruk) [inline apps/api/src/server/services/UserService.ts:75]: Doing this manually for now, but obviously we need a better to check auth on requests.

File: apps/api/src/server/services/UserService.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        createdAt: user.createdAt,
+        updatedAt: user.updatedAt,
+        lastLoginAt: user.lastLoginAt,
+      })
+      .from(user)
+      .where(eq(user.id, id));
+    return results.length !== 1 ? null : results[0];
+  }
+
+  public async updateProfile(id: number, updateData: ProfileData) {
```
[the remark this answers is not in the document]
Comment (tgrigoruk) [reply apps/api/src/server/services/UserService.ts:50]: @FoRVaiS I think you may have suggested a better way of doing this (?)

[the remark this answers is not in the document]
Comment (FoRVaiS) [reply apps/api/src/server/services/UserService.ts:50]: @tgrigoruk We can use the `UserType` from the schema.

The UserType will say that all/most of the keys are required but in the case for the PATCH, we can make everything optional: `Partial<UserType>`.

After importing `UserType`, we should end up with:
`type ProfileData = Partial<UserType>;`

**Review by FoRVaiS — APPROVED** [review]

**Pull request merged**